pkg/capabilities/consensus/ocr3/factory.go

```diff
@@ -21,35 +23,32 @@ type factory struct {
 	services.StateMachine
 }
 
-const (
-	// TODO(KS-617): read this from contract config
-	defaultMaxPhaseOutputBytes = 1000000 // 1 MB
-	defaultMaxReportCount      = 20
-)
-
-func newFactory(s *requests.Store, c *capability, batchSize int, outcomePruningThreshold uint64, lggr logger.Logger) (*factory, error) {
+func newFactory(s *requests.Store, c *capability, lggr logger.Logger) (*factory, error) {
 	return &factory{
-		store:                   s,
-		capability:              c,
-		batchSize:               batchSize,
-		outcomePruningThreshold: outcomePruningThreshold,
-		lggr:                    logger.Named(lggr, "OCR3ReportingPluginFactory"),
+		store:      s,
+		capability: c,
+		lggr:       logger.Named(lggr, "OCR3ReportingPluginFactory"),
 	}, nil
 }
 
-func (o *factory) NewReportingPlugin(ctx context.Context, config ocr3types.ReportingPluginConfig) (ocr3types.ReportingPlugin[[]byte], ocr3types.ReportingPluginInfo, error) {
-	rp, err := newReportingPlugin(o.store, o.capability, o.batchSize, config, o.outcomePruningThreshold, o.lggr)
-	info := ocr3types.ReportingPluginInfo{
+func (o *factory) NewReportingPlugin(_ context.Context, config ocr3types.ReportingPluginConfig) (ocr3types.ReportingPlugin[[]byte], ocr3types.ReportingPluginInfo, error) {
+	var configProto types.ReportingPluginConfig
+	err := proto.Unmarshal(config.OffchainConfig, &configProto)
+	if err != nil {
+		return nil, ocr3types.ReportingPluginInfo{}, err
+	}
```

Comment:
We need to apply defaults if any fields are set to 0. Otherwise we're risking a big failure during deployment.